### PR TITLE
fix(one line backtraces): handle correctly one line backtraces

### PR DESCRIPTION
**Pull request by juliayakovlev**
there are cases of backtraces in one line, we aren't handling them correctly

Example:
[shard 2] seastar - Exceptional future ignored: exceptions::mutation_write_timeout_exception
(Operation timed out for system.paxos - received only 0 responses from 1 CL=ONE.), backtrace:
0x3316f4d#012  0x2e2d177#012  0x189d397#012  0x2e76ea0#012  0x2e770af#012  0x2eaf065#012
0x2ebd68c#012  0x2e48d5d#012  /opt/scylladb/libreloc/libpthread.so.0+0x94e1#012

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [x] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [x] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1320,12 +1320,18 @@ def _read_system_log_and_publish_events(
...
                             LOGGER.debug(line)
                 match = backtrace_regex.search(line)
+                one_line_backtrace = []
                 if match and backtraces:
                     data = match.groupdict()
                     if data['other_bt']:
                         backtraces[-1]['backtrace'] += [data['other_bt'].strip()]
                     if data['scylla_bt']:
                         backtraces[-1]['backtrace'] += [data['scylla_bt'].strip()]
+                elif "backtrace:" in line.lower() and "0x" in line:
```
Comment (enaydanov) [inline sdcm/cluster.py:1330]: It's a common practice to add a comment with an example of a line which parsed here. Actually, this advice included in our Best Practices by @bentsi

File: unit_tests/test_cluster.py [hunk 2]
```diff
@@ -92,38 +92,58 @@ def test_search_system_log_specific_log(self):  # pylint: disable=invalid-name
...
 
-        events_file = open(EVENTS_PROCESSES['MainDevice'].raw_events_filename, 'r')
-        events = []
-        for line in events_file.readlines():
+        with open(EVENTS_PROCESSES['MainDevice'].raw_events_filename, 'r') as events_file:
+            events = []
+            for line in events_file.readlines():
 
-            events.append(json.loads(line))
+                events.append(json.loads(line))
```
Comment (enaydanov) [inline unit_tests/test_cluster.py:101]: The same thing done in another way later. Can you make it consistent? Also, you don't need call `fobj.readlines()` to read a file line-by-line. It's enough to use `for line in fobj`.

**Review by enaydanov — CHANGES_REQUESTED** [review]

File: sdcm/cluster.py (hunk 1 above)
Comment (juliayakovlev) [reply sdcm/cluster.py:1330]: Added

File: unit_tests/test_cluster.py (hunk 2 above)
Comment (juliayakovlev) [reply unit_tests/test_cluster.py]: Done

**Review by bentsi — APPROVED** [review]

**Comment by bentsi** [comment]
@juliayakovlev this is a new feature not a fix

**Comment by bentsi** [comment]
won't backport it

**Pull request merged**